src/rust/engine/src/nodes/mod.rs

```diff
@@ -553,6 +557,11 @@ impl Node for NodeKey {
         match self {
             NodeKey::Task(s) => s.task.cacheable,
             &NodeKey::SessionValues(_) | &NodeKey::RunId(_) => false,
+            NodeKey::PathMetadata(PathMetadataNode { subject_path, .. }) => match subject_path {
+                SubjectPath::Workspace(_) => true,
+                // TODO: Do not cache system path metadata lookups until we have filesystem watch support.
```

Comment:
Is this even worth caching in the Workspace or System cases? I'm not convinced that a cache lookup is faster than a single syscall.

Reply:
It depends in part on what the downstream uses of the metadata will do.

Just the metadata from the syscall? Then not worth caching probably. 

But if some process is re-run. That could have more of an impact.